contributors/design-proposals/pod-priority-resourcequota.md

```diff
@@ -0,0 +1,74 @@
+* ResourceQuota admission controller to check ResourceQuota priority class name and make corresponding decision.
+
+### Non-Goals 
+
+* Add priority in Pod spec (this is implemented in: #45610)
+
+## Background 
+
+Since we already have [priority filed in Pod spec](https://github.com/kubernetes/kubernetes/pull/45610), 
+Pods can now be classified into different priority classes. We would like to be able to create quota for various priority classes in order to manage cluster resources better and limit abuse scenarios. In order to implement this, we need to add a priority class name filed to ResourceQuota definition.
+
+## Overview 
+
+This design doc introduces a new field in ResourceQuota to specify a priority class and explains how quota enforcement logic is changed to apply the quota to pods with the given priority class.
+
+## Detailed Design 
+
+### Expected behavior of ResourceQuota
+
+The design will be like:
+
+```go
+// ResourceQuotaSpec defines the desired hard limits to enforce for Quota
+type ResourceQuotaSpec struct {
+  // Hard is the set of desired hard limits for each named resource
+  // +optional
+  Hard ResourceList
+  // A collection of filters that must match each object tracked by a quota.
+  // If not specified, the quota matches all objects.
+  // +optional
+  Scopes []ResourceQuotaScope
+  // If specified, indicates this ResourceQuota applies to that priority class
+  // If not specified, this ResourceQuota applies to pods with no PriorityClassName.
+  // +optional
+  PriorityClassName string
+}
+```
+PriorityClassName in ResourceQuota is matched against PriorityClassName in the Pod, including the case where PriorityClassName is not set. In detail:
+
+* One could expect when `PriorityClassName` field is set, this ResourceQuota object matches pods with the same priority class.
+* If this field is not present, it indicates this quota matches to pods that do not have a PriorityClassName (Pods' integer priorities would be ignored).
+
+It is worth noting that quota is assigned per namespace, not per user, in Kubernetes. Adding priority class name to ResourceQuota will not change anything with respect to the scope of ResourceQuota. In other words, when priority class name is defined for a ResourceQuota object, the quota is applied to all pods at that priority class in the specified namespace.
+
+Unlike pod spec, ResourceQuota does not have Priority *int32 in its spec.
+As explained above, ResourceQuota with a PriorityClassName matches against pods with the same PriorityClassName, regarless of the integer value of priority.
+
+### Expected behavior of ResourceQuota admission controller
+
+This part is about how to make admission controller work with priority aware `ResourceQuota`.
+
+* ResourceQuota admission controller should reject creation or update of ResourceQuota object if new ResourceQuota names a PriorityClass that does not exist, i.e. is not a pre-defined or user-specified PriorityClass (but empty is allowed).
```

Comment:
why?

Reply:
This is not strongly required but I am not sure if we want this flexibility. Would like to know your idea on this.

Reply:
it basically introduces ordering requirements.  i don't see a major value add by enforcing this constraint.

Reply:
Sure, let's remove this constraint.